src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1340,13 +1341,27 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return NULL;
 	}
 
-	if (!strcmp (m_class_get_name (cmethod->klass), "Vector256") || !strcmp (m_class_get_name (cmethod->klass), "Vector512"))
+	int vector_size;
+	if (!strcmp (m_class_get_name (cmethod->klass), "Vector64"))
+		vector_size = 64;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector128"))
+		vector_size = 128;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector256"))
+		vector_size = 256;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector512"))
+		vector_size = 512;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector"))
+		vector_size = 0;
```

Comment:
This is kind of mis-leading.

Reply:
Vector is actually a different class than Vector128/Vector256/ etc., it has different methods, so ideally, it should be handled in a separate function.

Reply:
Vector and Vector64/128/256 are now having the same sets of API's. There is no need to have a separate. Vector is an older class, comparing to Vector64/128/256.